diff --git a/main.py b/main.py
index d3708ac..b82b5d4 100644
--- a/main.py
+++ b/main.py
@@ -124,9 +124,9 @@ def train_loop(args, labeled_loader, unlabeled_loader, test_loader,
     unlabeled_iter = iter(unlabeled_loader)
 
     # for author's code formula
-    # moving_dot_product = torch.empty(1).to(args.device)
-    # limit = 3.0**(0.5)  # 3 = 6 / (f_in + f_out)
-    # nn.init.uniform_(moving_dot_product, -limit, limit)
+    moving_dot_product = torch.empty(1).to(args.device)
+    limit = 3.0**(0.5)  # 3 = 6 / (f_in + f_out)
+    nn.init.uniform_(moving_dot_product, -limit, limit)
 
     for step in range(args.start_step, args.total_steps):
         if step % args.eval_step == 0:
@@ -216,8 +216,8 @@ def train_loop(args, labeled_loader, unlabeled_loader, test_loader,
 
             # author's code formula
             # dot_product = s_loss_l_new - s_loss_l_old
-            # moving_dot_product = moving_dot_product * 0.99 + dot_product * 0.01
-            # dot_product = dot_product - moving_dot_product
+            moving_dot_product = moving_dot_product * 0.99 + dot_product * 0.01
+            dot_product = dot_product - moving_dot_product
 
             _, hard_pseudo_label = torch.max(t_logits_us.detach(), dim=-1)
             t_loss_mpl = dot_product * F.cross_entropy(t_logits_us, hard_pseudo_label)
